security/src/main/java/org/aggregatortech/dindora/security/authentication/providers/FileIDProvider.java

```diff
@@ -0,0 +1,107 @@
+package org.aggregatortech.dindora.security.authentication.providers;
+
+import com.fasterxml.jackson.core.JsonFactory;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.aggregatortech.dindora.common.ServiceLocatorHelper;
+import org.aggregatortech.dindora.common.io.system.SystemHelper;
+import org.aggregatortech.dindora.exceptions.IDStoreNotConfiguredException;
+import org.aggregatortech.dindora.exceptions.InvalidCredentialsException;
+import org.aggregatortech.dindora.exceptions.MessageService;
+import org.aggregatortech.dindora.security.authentication.token.AuthenticationCredentials;
+import org.aggregatortech.dindora.security.authentication.token.UserNamePasswordCredentials;
+import org.jvnet.hk2.annotations.Service;
+
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Optional;
+
+@Service
+public class FileIDProvider implements IDProvider {
+
+
+    public static final String IDSTORE_LOC = "idstore.loc";
+    //private JsonNode jsonNode;
+    private AuthenticationCredentials[] authCreds;
+
+
+    private File fileLoc = null;
+
+    @Override
+    public void configure() throws IDStoreNotConfiguredException {
+        //File fileLoc = null;
+        SystemHelper systemHelper = ServiceLocatorHelper.getServiceLocator().getService(SystemHelper.class);
```

Comment:
Please create a constructor and inject the servicelocator. Alternatively we can also inject the SystemHelper service directly.

Reply:
if you see this class implements interface and we want implementation class to specifically call configure method.

Reply:
This comment was about the mechanism to obtain the SystemHelper service. In general, I think we should let the framework auto-inject the dependencies. ServiceLocatorHelper should be used where auto-injection is not possible.